core/trino-main/src/main/java/io/trino/testing/MaterializedResult.java

```diff
@@ -457,6 +462,7 @@ public static Builder resultBuilder(ConnectorSession session, Iterable<? extends
         private final ConnectorSession session;
         private final List<Type> types;
         private final ImmutableList.Builder<MaterializedRow> rows = ImmutableList.builder();
+        private ImmutableList.Builder<String> columnNames = ImmutableList.builder();
```

Comment:
This can be List, no need for builder

in fact, this should be `Optional<List>`, as column names may be unknown